From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Thu, 11 Jul 2024 15:24:04 -0500
Subject: [PATCH] Force app_id to be set in more places

While testing the new version of kvirc I noticed that the setup window was using the `net.kvirc.kvirc` app_id. It seems my previous fix was incomplete since that window is started before the other window where we set the app_id.

Fix it by adding it to the setup window as well, and also just add it to a bunch more locations so that the chances of it not being set elsewhere are lower.
---
 src/kvilib/tal/KviTalApplication.cpp | 3 +++
 src/kvirc/ui/KviMainWindow.cpp       | 2 --
 src/kvirc/ui/KviWindow.cpp           | 3 +++
 src/modules/setup/SetupWizard.cpp    | 3 +++
 4 files changed, 9 insertions(+), 2 deletions(-)

diff --git a/src/kvilib/tal/KviTalApplication.cpp b/src/kvilib/tal/KviTalApplication.cpp
index 1878427a8..310eee757 100644
--- a/src/kvilib/tal/KviTalApplication.cpp
+++ b/src/kvilib/tal/KviTalApplication.cpp
@@ -29,6 +29,9 @@ KviTalApplication::KviTalApplication(int & iArgc, char ** ppcArgv)
 {
 	// Session management has been broken by source incompatible changes.
 	QObject::connect(this, SIGNAL(commitDataRequest(QSessionManager &)), this, SLOT(commitData(QSessionManager &)));
+
+	// set name of the app desktop file; used by wayland to load the window icon
+	QGuiApplication::setDesktopFileName("net.kvirc.KVIrc" KVIRC_VERSION_MAJOR);
 }
 
 KviTalApplication::~KviTalApplication()
diff --git a/src/kvirc/ui/KviMainWindow.cpp b/src/kvirc/ui/KviMainWindow.cpp
index 60d07d179..46ea0ff2e 100644
--- a/src/kvirc/ui/KviMainWindow.cpp
+++ b/src/kvirc/ui/KviMainWindow.cpp
@@ -105,8 +105,6 @@ KviMainWindow::KviMainWindow(QWidget * pParent)
 	// We try to avois this as much as possible, since it forces the use of the low-res 16x16 icon
 	setWindowIcon(*(g_pIconManager->getSmallIcon(KviIconManager::KVIrc)));
 #endif
-	// set name of the app desktop file; used by wayland to load the window icon
-	QGuiApplication::setDesktopFileName("net.kvirc.KVIrc" KVIRC_VERSION_MAJOR);

 	setWindowTitle(KVI_DEFAULT_FRAME_CAPTION);

diff --git a/src/kvirc/ui/KviWindow.cpp b/src/kvirc/ui/KviWindow.cpp
index 3224d7e78..c72caacc6 100644
--- a/src/kvirc/ui/KviWindow.cpp
+++ b/src/kvirc/ui/KviWindow.cpp
@@ -140,6 +140,9 @@ KviWindow::KviWindow(Type eType, const QString & szName, KviConsoleWindow * lpCo
 	setFocusPolicy(Qt::StrongFocus);
 	connect(g_pApp, SIGNAL(reloadImages()), this, SLOT(reloadImages()));
 
+	// set name of the app desktop file; used by wayland to load the window icon
+	QGuiApplication::setDesktopFileName("net.kvirc.KVIrc" KVIRC_VERSION_MAJOR);
+
 	setAttribute(Qt::WA_InputMethodEnabled, true);
 }
 
diff --git a/src/modules/setup/SetupWizard.cpp b/src/modules/setup/SetupWizard.cpp
index 67a7dcdba..86cfbe2dd 100644
--- a/src/modules/setup/SetupWizard.cpp
+++ b/src/modules/setup/SetupWizard.cpp
@@ -142,6 +142,9 @@ SetupWizard::SetupWizard()
 	QString szImagePath;
 	g_pApp->getGlobalKvircDirectory(szImagePath, KviApplication::Pics, "kvi_setup_label.png");
 
+	// set name of the app desktop file; used by wayland to load the window icon
+	QGuiApplication::setDesktopFileName("net.kvirc.KVIrc" KVIRC_VERSION_MAJOR);
+
 	m_pLabelPixmap = new QPixmap(szImagePath);
 	if(m_pLabelPixmap->isNull())
 	{
